contracts/FlowTransactionScheduler.cdc

```diff
@@ -1220,8 +1221,38 @@ access(all) contract FlowTransactionScheduler {
                         // charge the full fee for transaction execution
                         destroy tx.payAndRefundFees(refundMultiplier: 0.0)
 
-                        self.removeTransaction(txData: tx)
+                        // remove transaction object
+                        let transactionObject = self.transactions.remove(key: id)!
+
+                        transactionIDs.remove(key: id)
+
+                        if numRemoved >= self.config.collectionTransactionsLimit {
```

Comment:
I would just check this limit as first line in the inner for loop (before borrowTransaction) and if reached emit an even and actually return, so we leave to next itteration to clear remaining, this way its cleaner to me